sympy/polys/numberfields/galoisgroups.py

```diff
@@ -0,0 +1,622 @@
+
+    Explanation
+    ===========
+
+    Based on Alg 6.3.10 of [1], but uses resolvent coeff lookup.
+
+    """
+    from sympy.combinatorics.galois import S6TransitiveSubgroups
+
+    # First resolvent:
+
+    history = set()
+    for i in range(max_tries):
+        R_dup = get_resolvent_by_lookup(T, 1)
+        if dup_sqf_p(R_dup, ZZ):
+            break
+        _, T = tschirnhausen_transformation(T, max_tries=max_tries,
+                                            history=history,
+                                            fixed_order=not randomize)
+    else:
+        raise MaxTriesException
+
+    fl = dup_factor_list(R_dup, ZZ)
+
+    # Group the factors by degree.
+    factors_by_deg = defaultdict(list)
+    for r, _ in fl[1]:
+        factors_by_deg[len(r) - 1].append(r)
+
+    L = sorted(sum([
+        [d] * len(ff) for d, ff in factors_by_deg.items()
+    ], []))
+
+    T_has_sq_disc = has_square_disc(T)
+
+    if L == [1, 2, 3]:
+        f1 = factors_by_deg[3][0]
+        return ((S6TransitiveSubgroups.C6, False) if has_square_disc(f1)
+                else (S6TransitiveSubgroups.D6, False))
+
+    elif L == [3, 3]:
+        f1, f2 = factors_by_deg[3]
+        any_square = has_square_disc(f1) or has_square_disc(f2)
+        return ((S6TransitiveSubgroups.G18, False) if any_square
+                else (S6TransitiveSubgroups.G36m, False))
+
+    elif L == [2, 4]:
+        if T_has_sq_disc:
+            return (S6TransitiveSubgroups.S4p, True)
+        else:
+            f1 = factors_by_deg[4][0]
+            return ((S6TransitiveSubgroups.A4xC2, False) if has_square_disc(f1)
+                    else (S6TransitiveSubgroups.S4xC2, False))
+
+    elif L == [1, 1, 4]:
+        return ((S6TransitiveSubgroups.A4, True) if T_has_sq_disc
+                else (S6TransitiveSubgroups.S4m, False))
+
+    elif L == [1, 5]:
+        return ((S6TransitiveSubgroups.PSL2F5, True) if T_has_sq_disc
+                else (S6TransitiveSubgroups.PGL2F5, False))
+
+    elif L == [1, 1, 1, 3]:
+        return (S6TransitiveSubgroups.S3, False)
+
+    assert L == [6]
+
+    # Second resolvent:
+
+    history = set()
+    for i in range(max_tries):
+        R_dup = get_resolvent_by_lookup(T, 2)
+        if dup_sqf_p(R_dup, ZZ):
+            break
+        _, T = tschirnhausen_transformation(T, max_tries=max_tries,
+                                            history=history,
+                                            fixed_order=not randomize)
+    else:
+        raise MaxTriesException
+
+    T_has_sq_disc = has_square_disc(T)
+
+    if dup_irreducible_p(R_dup, ZZ):
+        return ((S6TransitiveSubgroups.A6, True) if T_has_sq_disc
+                else (S6TransitiveSubgroups.S6, False))
+    else:
+        return ((S6TransitiveSubgroups.G36p, True) if T_has_sq_disc
+                else (S6TransitiveSubgroups.G72, False))
+
+
+@public
+def galois_group(f, gens=None, args=None,
+                 by_name=False, max_tries=30, randomize=False):
```

Comment:
I wanted to ask advice on this. I'm trying to make the `galois_group` function fit into the pattern of functions that accept an `Expr`, convert it to a `Poly`, and then operate on that. One example would be,

https://github.com/sympy/sympy/blob/c4d6b3eb1c7b39832116f7b02baf524a1ffadfba/sympy/polys/polytools.py#L5839-L5843

There are many such functions, and they all seem to use the same signature, `(f, *gens, **args)`. Should I do the same?

What I have at the moment is kind of a mishmash. It makes `gens` and `args` into keyword args themselves. If I wanted to fit into the `(f, *gens, **args)` pattern, I would have to move the three special keyword args `by_name=False, max_tries=30, randomize=False` into `**args`.

I guess I'm struggling with this because I kind of see "two worlds" colliding here: there's the world of functions using the `(f, *gens, **args)` signature -- which I think are probably on the older side -- and there's the world of functions that are trying to follow the (newer?) rules for writing docstrings, in particular, [the rules about the parameters section](https://docs.sympy.org/latest/contributing/docstring.html#parameters-section).

Do we need rules for documenting `**kwargs` dictionaries?

In a case like this, I would want to document the three kwargs `by_name=False, max_tries=30, randomize=False` that are special to this function, and then somehow say that, "anything else in `**args` will be passed onward to the `poly_from_expr()` function; see that function for further details." I don't know if there's a protocol for this yet. I know there has been some discussion (#23376) about trying to avoid use of `**kwargs` where possible.

Reply:
Yeah, good question, and I had similar thoughts about the poly options -- they don't serve much purpose here.

Maybe I can just omit `*gens` and `**args` completely? If the user passes a constant expression, that's just going to result in an error anyway, so I don't see any need for `*gens`.

Reply:
For a constant polynomial would it not be the trivial group?

Reply:
A case could probably be made either way. What's for sure is that it's one of those funny, degenerate cases. I think my thinking was something like, "What's the group of all permutations of an empty set of roots? Should be an empty group, since there aren't any ways to permute no roots?"

Alternatively, there's "one way," to permute no roots, namely, to "do nothing," which is what the identity permutation does. : )

I guess in practical terms it doesn't matter much.

Reply:
Yeah, going back to the actual definition instead of trying to reason about "permutations of nothing" is certainly a good approach! And I did think about it that way as well, but wound up asking myself whether constant polynomials are regarded as even having splitting fields. I guess it may depend on the author. My source of truth on this stuff has always been Gallian's algebra, and when I went back there the answer was no.

Maybe the key is in the word "splitting," which refers to factorization, so it's my fault for introducing the subject of roots at all. As you point out, we should be thinking about polynomials over a field $K$. In that case, in all questions about factoring and irreducibility, constant polynomials are either zero, or units in the ring $K[x]$, and thus not even the sorts of things for which factorization is a question.

Gallian's system seems like a sensible one: $f(x) \in K[x]$ *splits* in an extension $L/K$ if it can be factored as a product of linear factors in $L[x]$. $L$ is a splitting field for $f(x)$ over $K$ if $f(x)$ splits in $L$ but in no proper subfield of $L$.